components/StyledTextarea.js

```diff
@@ -127,10 +128,66 @@ export default class StyledTextarea extends React.PureComponent {
     }
   };
 
-  render() {
-    const { autoSize, showCount, resize, ...props } = this.props;
+  renderCountMessage() {
+    const { showCount, ...props } = this.props;
     const value = props.value || props.defaultValue || '';
 
+    if (!showCount || props.disabled || props.readOnly) {
+      return '';
+    }
+
+    let formattedMessage;
+    if (props.maxLength) {
+      const charactersRemaining = props.maxLength - value.length;
+      if (charactersRemaining > 1) {
+        formattedMessage = (
+          <FormattedMessage
+            id="textarea.charactersRemaining"
+            defaultMessage="Remaining {charactersRemaining} characters."
+            values={{ charactersRemaining: charactersRemaining }}
+          />
+        );
+      } else {
+        formattedMessage = (
+          <FormattedMessage
+            id="textarea.characterRemaining"
+            defaultMessage="Remaining {charactersRemaining} character."
+            values={{ charactersRemaining: charactersRemaining }}
+          />
+        );
+      }
```

Comment:
The way to go for these cases is [plural](https://formatjs.io/docs/core-concepts/icu-syntax/#plural-format). Also, `id` is optional.

```suggestion
      formattedMessage = (
          <FormattedMessage
            defaultMessage="Remaining {charactersRemaining, plural, one {# character} other {# characters}}."
            values={{ charactersRemaining: props.maxLength - value.length }}
          />
      )
```